src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -127,8 +127,8 @@ private static AnalysisResults analyzeImage(ByteString imageBytes)
     AnalysisResults.Builder analysisBuilder = retrieveText(imageBytes);
     ImmutableSet<String> categories = categorizeText(analysisBuilder.getRawText());
 
-    Stream<String> tokens = getTokensFromRawText(analysisBuilder.getRawText());
-    checkForParsableDate(analysisBuilder, tokens);
+    checkForParsableDate(analysisBuilder);
```

Comment:
I feel it's more readable if you pass in the raw text into checkForParsableDate() and have it return the timestamp. Same on the line below.

Reply:
I wrote it this way to avoid the extra complexity involved in returning Optionals. Since it's an private helper method, I don't think the interface is all that important, hence why I focused on simplicity instead.